src/java/org/apache/cassandra/db/virtual/DescribeTables.java

```diff
@@ -0,0 +1,306 @@
+
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Streams;
+
+import org.apache.cassandra.cql3.functions.Function;
+import org.apache.cassandra.cql3.functions.FunctionName;
+import org.apache.cassandra.cql3.functions.UDAggregate;
+import org.apache.cassandra.cql3.functions.UDFunction;
+import org.apache.cassandra.db.DecoratedKey;
+import org.apache.cassandra.db.Keyspace;
+import org.apache.cassandra.db.SchemaCQLHelper;
+import org.apache.cassandra.db.marshal.CompositeType;
+import org.apache.cassandra.db.marshal.UTF8Type;
+import org.apache.cassandra.db.marshal.UserType;
+import org.apache.cassandra.dht.LocalPartitioner;
+import org.apache.cassandra.schema.Functions;
+import org.apache.cassandra.schema.IndexMetadata;
+import org.apache.cassandra.schema.Schema;
+import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.schema.Types;
+import org.apache.cassandra.schema.ViewMetadata;
+import org.apache.cassandra.utils.ByteBufferUtil;
+
+public class DescribeTables
+{
+    private static final String KEYSPACE = "keyspace_name";
+    private static final String CQL = "cql";
+
+    private static final CompositeType utfComposite = CompositeType.getInstance(UTF8Type.instance, UTF8Type.instance);
+
+    public static Collection<VirtualTable> getAll(String name)
+    {
+        return ImmutableList.of(new DescribeKeyspaceTable(name),
+                                new DescribeIndexesTable(name),
+                                new DescribeTypesTable(name),
+                                new DescribeAggregatesTable(name),
+                                new DescribeFunctionsTable(name),
+                                new DescribeViewsTable(name),
+                                new DescribeTablesTable(name));
+    }
+
+    static final class DescribeKeyspaceTable extends AbstractVirtualTable
+    {
+        DescribeKeyspaceTable(String keyspace)
+        {
+            super(TableMetadata.builder(keyspace, "describe_keyspace")
+                               .comment("cql for keyspace metadata")
+                               .kind(TableMetadata.Kind.VIRTUAL)
+                               .partitioner(new LocalPartitioner(UTF8Type.instance))
+                               .addPartitionKeyColumn(KEYSPACE, UTF8Type.instance)
+                               .addRegularColumn(CQL, UTF8Type.instance)
+                               .build());
+        }
+
+        @Override
+        public DataSet data(DecoratedKey partitionKey)
+        {
+            String keyspace = UTF8Type.instance.compose(partitionKey.getKey());
+
+            SimpleDataSet result = new SimpleDataSet(metadata());
+            result.row(keyspace)
+                  .column(CQL, SchemaCQLHelper.getKeyspaceAsCQL(keyspace));
```

Comment:
This surfaces a ServerError with an NPE if the keyspace doesn't exist, instead of returning 0 rows:

```
ERROR [Native-Transport-Requests-1] 2019-09-04 23:05:26,228 QueryMessage.java:124 - Unexpected error during query
java.lang.NullPointerException: null
        at org.apache.cassandra.db.SchemaCQLHelper.toCQL(SchemaCQLHelper.java:152)
        at org.apache.cassandra.db.SchemaCQLHelper.getKeyspaceAsCQL(SchemaCQLHelper.java:65)
        at org.apache.cassandra.db.virtual.DescribeTables$DescribeKeyspaceTable.data(DescribeTables.java:88)
```